ports/esp32/Makefile

```diff
@@ -347,6 +347,7 @@ SRC_C = \
 	modesp32.c \
 	espneopixel.c \
 	machine_hw_spi.c \
+	machine_i2s.c \
```

Comment:
Note for anyone willing to compile their own build from `master` branch and merging in this PR:
The Makefile has changed a lot on `master`, and this line should now make it to `ports/esp32/main/CMakeLists.txt` in the `set(MICROPY_SOURCE_PORT ...)` instruction:

```
set(MICROPY_SOURCE_PORT
    ...
    ${PROJECT_DIR}/machine_i2s.c
)

```

Reply:
thanks for flagging this.  I updated the PR to work with the new cmake build

Reply:
Thank you for the amazing work!
Being able to compile the firmware and have sound flowing through my I2S to a speaker in record time was quite a joyful moment ^_^